packages/graphql_codegen/test/assets/when_example/test.dart

```diff
@@ -0,0 +1,49 @@
+import 'document.graphql.dart';
+
+void printWithWhen(Query$Query$u? union) {
+  union?.when(
+    tA: (tA) => print("tA"),
+    tB: (tB) => print("tB"),
+    orElse: () => print("orElse"),
+  );
+}
+
+void returnAndPrintWithWhen(Query$Query$u? union) {
+  final result = union?.when(
+    tA: (tA) => "tA",
+    tB: (tB) => "tB",
+    orElse: () => "orElse",
+  );
+  print(result);
+}
+
+void printAllWithMaybeWhen(Query$Query$u? union) {
+  union?.maybeWhen(
+    tA: (tA) => print("tA"),
+    tB: (tB) => print("tB"),
+    orElse: () => print("orElse"),
+  );
+}
+
+void printSomeWithMaybeWhen(Query$Query$u? union) {
+  union?.maybeWhen(
+    tA: (tA) => print("tA"),
+    orElse: () => print("tB, orElse"),
+  );
+}
+
+void returnAndPrintWithMaybeWhen(Query$Query$u? union) {
+  final result = union?.maybeWhen(
+    tA: (tA) => "tA",
+    orElse: () => "tB, orElse",
+  );
+  print(result);
+}
+
+void printQuery(Query$Query q) {
+  printWithWhen(q.u);
+  returnAndPrintWithWhen(q.u);
+  printAllWithMaybeWhen(q.u);
+  printSomeWithMaybeWhen(q.u);
+  returnAndPrintWithMaybeWhen(q.u);
+}
```

Comment:
@budde377 is this the right way to add a new test? 

I tried to think of the main things folks would try to do with this feature and capture them here. I ended up with 3 things: 
1. use when/maybeWhen with a void return type
2. use when/maybeWhen with a non-void return type
3. use mabyeWhen exhaustively
4. use maybeWhen non-exhaustively

Reply:
I just realized I've shared a pretty poor example of a test. Try and have a look at this file instead: https://github.com/heftapp/graphql_codegen/blob/ae0763ab6fcaaafd1203d35191dba19ff47d8faf/packages/graphql_codegen/test/assets/issue_192/serializer_test.dart#L2

Reply:
ah haha okay, i was like, "is there some magic here or are we just testing that things can be compiled?"

Reply:
done!